runtime/util/vmargs.c

```diff
@@ -578,8 +578,17 @@ optionValueOperations(J9PortLibrary *portLibrary, J9VMInitArgs* j9vm_args, IDATA
 					switch (*cursor) {
 						case '\0':
 							oldValue = value;
-							value = (value +  sizeof(UDATA) - 1) & ~(sizeof(UDATA) - 1);		/* round to nearest pointer value */
-							if (value < oldValue) return OPTION_OVERFLOW;
+							/* Round up to the next multiple of the pointer width */
+							value = (value + sizeof(UDATA) - 1) & ~(sizeof(UDATA) - 1);
+
+							if ((value < oldValue) || (((I_64) value) < ((I_64) oldValue))) {
+								/*
+								 * Truncate (round in a negative direction) when overflow is detected
+								 * in either the signed 64-bit or default unsigned representations.
+								 * This avoids sign issues regardless of downstream usage.
+								 */
+								value = oldValue & ~(sizeof(UDATA) - 1);
+							}
 							break;
```

Comment:
I don't think this has the desired behavior for 32-bit platforms.
```c
#if defined(J9VM_ENV_DATA64)
#define MEM_MAX_LIMIT I_64_MAX
#else /* defined(J9VM_ENV_DATA64) */
#define MEM_MAX_LIMIT U_32_MAX
#endif /* defined(J9VM_ENV_DATA64) */
							/* Round (up if possible) to a multiple of the pointer width. */
							if (value < ((UDATA)MEM_MAX_LIMIT & ~(sizeof(UDATA) - 1))) {
								value += sizeof(UDATA) - 1;
							}
							value &= ~(sizeof(UDATA) - 1);
#undef MEM_MAX_LIMIT
							break;
```

Reply:
It applies a limit of 2^32-1 on 32-bit platforms and 2^63-1 on 64-bit platforms.
I would argue it's also easier to follow.

Reply:
Also, on 32-bit platforms, `UDATA` and `I_64` differ both in width and signedness. A cast from `UDATA` to `I_64` is not well defined: Does the width change first and then get sign-extended or the other way around?